services/plugins/npm-plugins-source.ts

```diff
@@ -0,0 +1,48 @@
+import {PluginsSourceBase} from "./plugins-source-base";
+import Future = require("fibers/future");
+
+export class NpmPluginsSource extends PluginsSourceBase implements IPluginsSource {
+	constructor(private $childProcess: IChildProcess,
+		private $hostInfo: IHostInfo,
+		private $npmService: INpmService,
+		private $progressIndicator: IProgressIndicator,
+		private $logger: ILogger,
+		private $errors: IErrors) {
+		super();
+	}
+
+	public initialize(projectDir: string, keywords: string[]): IFuture<void> {
+		return (() => {
+			super.initialize(projectDir, keywords).wait();
+
+			let searchFuture = this.$npmService.search(this._projectDir, keywords);
+
+			this.$logger.printInfoMessageOnSameLine("Searching for plugins in npm, please wait.");
+
+			this.$progressIndicator.showProgressIndicator(searchFuture, 2000).wait();
+
+			this._plugins = searchFuture.isResolved() ? searchFuture.get() : searchFuture.wait();
```

Comment:
`showProgressIndicator`'s implementation `wait`s for the future passed as a parameter. Is there a reason you check if it's resolved here instead of directly getting the result?